js/adapt-contrib-trickle.js

```diff
@@ -1,258 +1,558 @@
+            }
+
+            this.resizeBodyToCurrentIndex();
+            this._listenToResizeEvent = true;
+        },
+
+        onRemove: function(view) {
+            this.endTrickle();
+        },
+
+
+        model: new Backbone.Model({}),
+
+        _listenToResizeEvent: false,
+        _isPageInitialized: false,
+        _isFinished: false,
+        _currentStepIndex: 0,
+        _descendantsChildrenFirst: null,
+        _descendantsParentFirst: null,
+        _pageView: null,
+
+        initialize: function() {
+            this.listenToOnce(Adapt, "app:dataReady", this.onDataReady);
+        },
+
+        setupEventListeners: function() {
+            this._onWrapperResize = _.bind(Trickle.onWrapperResize, Trickle);
+            $("#wrapper").on('resize', this._onWrapperResize );
+
+            this.listenTo(Adapt, "remove", this.onRemove);
+            this.listenTo(Adapt, "pageView:preRender", this.onPagePreRender);
+            this.listenTo(Adapt, "pageView:postRender", this.onPagePostRender);
+            this.listenTo(Adapt, "pageView:ready", this.onPageReady);
+
+            this.listenTo(Adapt, "articleView:preRender", this.onArticlePreRender);
+            this.listenTo(Adapt, "blockView:postRender articleView:postRender", this.onArticleAndBlockPostRender);
+
+            this.listenTo(Adapt.articles, "change:"+completionAttribute, this.onAnyComplete);
+            this.listenTo(Adapt.blocks, "change:"+completionAttribute, this.onAnyComplete);
+            this.listenTo(Adapt.components, "change:"+completionAttribute, this.onAnyComplete);           
+
+            this.listenTo(Adapt, "trickle:interactionComplete", this.onAnyComplete);
+
+            this.listenTo(Adapt, "steplocking:wait", this.onStepUnlockWait);
+            this.listenTo(Adapt, "steplocking:unwait", this.onStepUnlockUnwait);
+
+            this.listenTo(Adapt, "trickle:relativeScrollTo", this.relativeScrollTo);
+        },
+
+        initializePage: function(view) {
+            var pageId = view.model.get("_id");
+
+            var pageConfig = Adapt.course.get("_trickle");
+            if (pageConfig && pageConfig._isEnabled === false) return;
+
+            this._descendantsChildrenFirst =  Models.getDescendantsFlattened(pageId);
+            this._descendantsParentFirst = Models.getDescendantsFlattened(pageId, true);
+            this._currentStepIndex = 0;
+            this._isFinished = false;
+            this._listenToResizeEvent = false;
+            this._pageView = view;
+
+            this.checkResetChildren();
+
+            this.initializeStepUnlockWait();
+
+            this._isPageInitialized = true;
+
+        },
+
+        checkResetChildren: function() {
+            var descendantsChildrenFirst = this._descendantsChildrenFirst;
+            for (var i = 0, model; model = descendantsChildrenFirst.models[i++];) {
+                this.checkResetModel(model);
+            }
+        },
+
+        checkResetModel: function(model) {
```

Comment:
no, of the checkResetModel function. sorry, Github gives you a really odd preview there, the comment was actually related to L147